Src/ILGPU/Runtime/CPU/CPUMemoryBuffer.cs

```diff
@@ -38,34 +38,37 @@ public sealed class CPUMemoryBuffer<T, TIndex> : MemoryBuffer<T, TIndex>
         internal CPUMemoryBuffer(CPUAccelerator accelerator, TIndex extent)
             : base(accelerator, extent)
         {
-            NativePtr = Marshal.AllocHGlobal(extent.Size * Interop.SizeOf<T>());
+            NativePtr = Marshal.AllocHGlobal(
+                new IntPtr(extent.Size * Interop.SizeOf<T>()));
         }
 
         #endregion
 
         #region Methods
 
         /// <summary cref="MemoryBuffer{T, TIndex}.CopyToView(
-        /// AcceleratorStream, ArrayView{T}, Index1)"/>
+        /// AcceleratorStream, ArrayView{T}, LongIndex1)"/>
         protected internal unsafe override void CopyToView(
             AcceleratorStream stream,
             ArrayView<T> target,
-            Index1 sourceOffset)
+            LongIndex1 sourceOffset)
         {
             var binding = stream.BindScoped();
 
             var sourceAddress = ComputeEffectiveAddress(sourceOffset);
+            var targetAddress = target.LoadEffectiveAddress();
             switch (target.AcceleratorType)
             {
                 case AcceleratorType.CPU:
-                    Unsafe.CopyBlock(
-                        target.LoadEffectiveAddress(),
+                    Buffer.MemoryCopy(
                         sourceAddress,
-                        (uint)target.LengthInBytes);
+                        targetAddress,
+                        target.LengthInBytes,
+                        target.LengthInBytes);
```

Comment:
QUESTION: Is this copy going to overrun the source buffer?

Reply:
I think this should not happen as `Buffer.MemoryCopy` seems to have the semantics of a safe `memcpy` function in C.

Reply:
I mean that `sourceAddress = ComputeEffectiveAddress(sourceOffset);`. Then, `sourceAddress + target.LengthInBytes` may read beyond `sourceAddress + MemoryBuffer.LengthInBytes`.

Reply:
I have analyzed this piece of code, and it should not happen that overruns the source buffer, since all externally visible methods perform boundary checks. The previous piece of code also performed a simple block copy into the target buffer.